x/ecocredit/server/basket/msg_create.go

```diff
@@ -16,25 +17,35 @@ import (
 // Create is an RPC to handle basket.MsgCreate
 func (k Keeper) Create(ctx context.Context, msg *basket.MsgCreate) (*basket.MsgCreateResponse, error) {
 	sdkCtx := sdk.UnwrapSDKContext(ctx)
+
 	var fee sdk.Coins
 	k.paramsKeeper.Get(sdkCtx, core.KeyBasketCreationFee, &fee)
-	if err := basket.ValidateMsgCreate(msg, fee); err != nil {
-		return nil, err
+	if !msg.Fee.IsAllGTE(fee) {
+		return nil, sdkerrors.ErrInsufficientFee.Wrapf("minimum fee %s, got %s", fee, msg.Fee)
```

Comment:
Moved from `msg_create.go` (stateless message validation) and included directly.